bin/bento

```diff
@@ -331,6 +508,296 @@ class NormalizeRunner
+    time = Benchmark.measure do
+      metadata_files.each do |metadata_file|
+        m = box_metadata(metadata_file)
+        destroy_all_bento
+        test_box(m['name'], m['providers'])
+        destroy_all_bento
+      end
+    end
+    banner("Testing finished in #{duration(time.real)}.")
+  end
+
+  private
+
+  def destroy_all_bento
+    cmd = Mixlib::ShellOut.new("vagrant box list | grep 'bento-'")
+    cmd.run_command
+    boxes = cmd.stdout.split("\n")
+
+    boxes.each do |box|
+      b = box.split(' ')
+      rm_cmd = Mixlib::ShellOut.new("vagrant box remove --force #{b[0]} --provider #{b[1].to_s.gsub(/(,|\()/, '')}")
+      banner("Removing #{b[0]} for provider #{b[1].to_s.gsub(/(,|\()/, '')}")
+      rm_cmd.run_command
+    end
+  end
+
+  def test_box(boxname, providers)
+    providers.each do |provider, provider_data|
+
+      plugin_list = Mixlib::ShellOut.new("vagrant plugin list | awk '{print $1}'")
+      plugin_list.run_command
+      plugins = plugin_list.stdout.split("\n")
+
+      if provider == 'vmware_desktop'
+        case
+        when plugins.include?('vagrant-vmware-workstation')
+          provider = 'vmware_workstation'
+        when plugins.include?('vagrant-vmware-fusion')
+          provider = 'vmware_fusion'
+        end
+      end
+
+      share_disabled = shared_folder ? false : true
+
+      kitchen_cfg = {"provisioner"=>{"name"=>"chef_zero", "data_path"=>"test/fixtures"},
+      "platforms"=>
+        [{"name"=>"#{boxname}-#{provider}",
+          "driver"=>
+          {"name"=>"vagrant",
+            "synced_folders"=>[[".", "/vagrant", "disabled: #{share_disabled}"]],
+            "provider"=>provider,
+            "box"=>"bento-#{boxname}",
+            "box_url"=>"file://#{ENV['PWD']}/builds/#{provider_data['file']}"}}],
+      "suites"=>[{"name"=>"default", "run_list"=>nil, "attributes"=>{}}]}
+
+      File.open(".kitchen.#{provider}.yml", "w") { |f| f.write(kitchen_cfg.to_yaml) }
+
+      Kitchen.logger = Kitchen.default_file_logger
+      @loader = Kitchen::Loader::YAML.new(project_config: "./.kitchen.#{provider}.yml")
+      config = Kitchen::Config.new(loader: @loader)
+      config.instances.each do |instance|
+        instance.test(:always)
+      end
+    end
+  end
+end
+
+class DeleteRunner
+  include Common
+  include HttpStuff
+
+  attr_reader :boxname, :version
+
+  def initialize(opts)
+    @boxname = opts.box
+    @version = opts.version
+  end
+
+  def start
+    banner("Starting Delete...")
+    time = Benchmark.measure do
+      delete_version(boxname, version)
+    end
+    banner("Delete finished in #{duration(time.real)}.")
+  end
+
+  private
+
+  def delete_version(boxname, version)
+    banner("Deleting version #{version} of box #{boxname}")
+    req = request('delete', "#{atlas_api}/box/#{atlas_org}/#{boxname}/version/#{version}", { 'access_token' => atlas_token }, { 'Content-Type' => 'application/json' })
+
+    case req.code
+    when '200'
```

Comment:
Codes 200-399 are all valid responses, where 3xx is usually some sort of redirect.